core/Piwik.php

```diff
@@ -968,4 +968,21 @@ public static function getEarliestDateToRearchive()
 
         return Date::yesterday()->subMonth($lastNMonthsToInvalidate)->setDay(1);
     }
+
+    /**
+     * Given the fully qualified name of a class located within a Matomo plugin,
+     * returns the name of the plugin.
+     *
+     * Uses the fact that Matomo plugins have namespaces like Piwik\Plugins\MyPlugin.
+     *
+     * @param string $className the name of a class located within a Matomo plugin
+     * @return string the plugin name
+     */
+    public static function getPluginNameOfMatomoClass(string $className): string
```

Comment:
This would work for any non-Matomo plugins as well, right?
```suggestion
    public static function getPluginNameFromClass(string $className): string
```

Reply:
it would only work for a class ina matomo plugin, it would not work for any class